### PR TITLE
docs: update chisel how-to guides

### PR DESCRIPTION
**Pull request by linostar**
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----
This update chisel how-to guides to include details such as the new command `chisel find` and chisel-releases' integrations tests for contributors.

### CODE REVIEW

File: docs/how-to/chisel/publish-slice.rst [hunk 1]
```diff
@@ -34,10 +34,18 @@ It is as simple as proposing your changes into the upstream
 
    cp <path/to/your/slice/definitions/file> slices
    git add slices/
-   git commit -m "feat: add new slice definitions for 'name_of_the_package'"
+   git commit -m "feat(22.04): add new slice definitions for 'package_name'"
    git push origin create-openssl-bins-slice
 
-Create a pull request and wait for it to be merged.
+4. Create integration tests to test your changes.
```
Comment (lengau) [inline docs/how-to/chisel/publish-slice.rst:40]: Nice!

**Review by lengau — APPROVED** [review]
This is a good improvement. I do wonder though whether we should be considering where this whole document belongs compared to https://github.com/canonical/chisel-releases/blob/main/CONTRIBUTING.md

Specifically, this slicing how-to seems more about chisel than about Rockcraft. However, it also seems like chisel doesn't have an equivalent docs site that we could move this to.

File: docs/how-to/chisel/publish-slice.rst [hunk 2]
```diff
@@ -34,10 +34,18 @@ It is as simple as proposing your changes into the upstream
...
+If necessary, add integration tests for your changes in
+``tests/spread/integration/<package_name>``. Those tests are composed of a
+``task.yaml`` file (plus a few accompanying test scripts if needed) that
+contains instructions on how to cut the slices you created or changed and test
+their functionalities. You can find examples on how to write such tests in the
+children folders of ``tests/spread/tests``.
```
Comment (medubelko) [inline docs/how-to/chisel/publish-slice.rst:46]:
```suggestion
child folders of ``tests/spread/tests``.
```

**Review by medubelko — CHANGES_REQUESTED** [review]

**Comment by medubelko** [comment]
> I do wonder though whether we should be considering where this whole document belongs compared to https://github.com/canonical/chisel-releases/blob/main/CONTRIBUTING.md

Medium-term, I'd like all project contributor material to be in README and CONTRIBUTING, and platform/publishing/authoring material in the docs. Does that split make sense for how we've set up chisel?

As for documentation between microprojects, that will have to remain a longer-term concern.

**Review by cjdcordeiro — APPROVED** [review]

**Review by tigarmo — APPROVED** [review]

**Review by medubelko — APPROVED** [review]

**Comment by linostar** [comment]
@tigarmo This ready for merging. The failing spread tests are due to the problem in LXD waiting for a fix.

**Pull request merged**